optuna/integration/dask.py

```diff
@@ -0,0 +1,667 @@
+        return self.get_storage(storage_name).set_trial_state_values(
+            trial_id=trial_id,
+            state=TrialState[state],
+            values=values,
+        )
+
+    def set_trial_intermediate_value(
+        self,
+        comm: "distributed.comm.tcp.TCP",
+        storage_name: str,
+        trial_id: int,
+        step: int,
+        intermediate_value: float,
+    ) -> None:
+        return self.get_storage(storage_name).set_trial_intermediate_value(
+            trial_id=trial_id,
+            step=step,
+            intermediate_value=intermediate_value,
+        )
+
+    def set_trial_user_attr(
+        self,
+        comm: "distributed.comm.tcp.TCP",
+        storage_name: str,
+        trial_id: int,
+        key: str,
+        value: Any,
+    ) -> None:
+        return self.get_storage(storage_name).set_trial_user_attr(
+            trial_id=trial_id,
+            key=key,
+            value=value,
+        )
+
+    def set_trial_system_attr(
+        self,
+        comm: "distributed.comm.tcp.TCP",
+        storage_name: str,
+        trial_id: int,
+        key: str,
+        value: Any,
+    ) -> None:
+        return self.get_storage(storage_name).set_trial_system_attr(
+            trial_id=trial_id,
+            key=key,
+            value=value,
+        )
+
+    def get_trial(
+        self,
+        comm: "distributed.comm.tcp.TCP",
+        storage_name: str,
+        trial_id: int,
+    ) -> dict:
+        trial = self.get_storage(storage_name).get_trial(trial_id=trial_id)
+        return _serialize_frozentrial(trial)
+
+    def get_all_trials(
+        self,
+        comm: "distributed.comm.tcp.TCP",
+        storage_name: str,
+        study_id: int,
+        deepcopy: bool = True,
+        states: Optional[Tuple[str, ...]] = None,
+    ) -> List[dict]:
+        deserialized_states = None
+        if states is not None:
+            deserialized_states = tuple(TrialState[s] for s in states)
+        trials = self.get_storage(storage_name).get_all_trials(
+            study_id=study_id,
+            deepcopy=deepcopy,
+            states=deserialized_states,
+        )
+        return [_serialize_frozentrial(t) for t in trials]
+
+    def get_n_trials(
+        self,
+        comm: "distributed.comm.tcp.TCP",
+        storage_name: str,
+        study_id: int,
+        state: Optional[TrialState] = None,
+    ) -> int:
+        return self.get_storage(storage_name).get_n_trials(
+            study_id=study_id,
+            state=state,
+        )
+
+
+def _register_with_scheduler(
+    dask_scheduler: "distributed.Scheduler", storage: Union[None, str, BaseStorage], name: str
+) -> None:
+    if "optuna" not in dask_scheduler.extensions:
+        ext = _OptunaSchedulerExtension(dask_scheduler)
+    else:
+        ext = dask_scheduler.extensions["optuna"]
+
+    if name not in ext.storages:
+        ext.storages[name] = optuna.storages.get_storage(storage)
+
+
+@experimental_class("3.0.4")
```

Comment:
Could you update the version to v3.1.0, which is our next minor release?

```suggestion
@experimental_class("3.1.0")
```

FYI:Optuna v3.1 will be released in mid-January 2023 with the following new features.
https://github.com/optuna/optuna/discussions/4064